fuzzbench/run_experiment.py

```diff
@@ -17,37 +17,72 @@
 
 import redis
 import rq
+from rq.job import Job
 
+from common import config_utils, environment, yaml_utils
+from experiment.build import docker_images
 from fuzzbench import jobs
 
+redis_connection = redis.Redis(host="queue-server")  # pylint: disable=invalid-name
 
-def run_experiment():
+
+def run_experiment(config):
     """Main experiment logic."""
     print('Initializing the job queue.')
     queue = rq.Queue()
+
+    images_to_build = docker_images.get_images_to_build(config['fuzzers'],
+                                                        config['benchmarks'])
     jobs_list = []
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-image',
-                      job_timeout=600,
-                      job_id='base-image'))
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-builder',
-                      job_timeout=600,
-                      job_id='base-builder',
-                      depends_on='base-image'))
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-runner',
-                      job_timeout=600,
-                      job_id='base-runner',
-                      depends_on='base-image'))
+    unqueued_build_images = []
+    for name, obj in images_to_build.items():
+        if name in ['base-image']:
+            jobs_list.append(
+                queue.enqueue(jobs.build_image,
+                              tag=obj['tag'],
+                              context=obj['context'],
+                              job_timeout=600,
+                              job_id=name))
+            continue
+
+        if len(obj['depends_on']) > 1:
+            unqueued_build_images.append((name, obj))
+            continue
```

Comment:
Since `rq` doesn't support multi-dependency and the discussion has been existing for 7 years, I choose a simple way to implement this.
Simply, log those multi-depend jobs down, and add them during an iteration where checking whether all its depended jobs finish.

Any other thoughts? Or better suggestions?

Reply:
We resolve this by removing multi-dependency.